dev/io.openliberty.microprofile.reactive.messaging.internal_fat.common/src/com/ibm/ws/microprofile/reactive/messaging/fat/kafka/containers/ExtendedKafkaContainer.java

```diff
@@ -111,8 +145,45 @@ public ExtendedKafkaContainer withKafkaJaasConfig(String kafkaJaasConfig) {
         return this;
     }
 
+    /**
+     * Get Primary Keystore file
+     *
+     * If TLS has not been enabled returns null
+     *
+     * The contents of this keystore is used as the basis of trust and key stores for both Liberty and Kafka
+     * therefore its contents should always be trusted if used on either side of the communication
+     *
+     * This will return null if at least TLS is not enabled on the kafka definition
+     *
+     * @return
+     */
     public File getKeystoreFile() {
-        return new File(KEYSTORE_FILENAME);
+        if(tls) {
+            return new File(KEYSTORE_FILENAME);
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Gets a secondary Keystore file if MTLS has been enabled, otherwise returns null
+     *
+     * This should be used for Client keystores such as RM Channels or connector
+     *
+     * By default, this keystore is not included in the truststores of either Liberty or Kafka. If used as the keystore
+     * for either server or client sides should result in a SSL Handshake error
+     *
+     * If you want to include the contents of this keystore in the truststore used for kafka, then `mergeKeyStores()` must be used
+     * to created a merged server truststore.
+     *
+     * @return
+     */
+    public File getKeystoreFile2() {
+        if(mtls) {
+            return new File(KEYSTORE_FILENAME2);
+        } else {
+            return null;
+        }
```

Comment:
I don't like this because it doesn't match how we expect users to deploy mTLS and it's hard to tell what's in each keystore and where it should be used.

We should re-write how we create the key and trust stores and ensure they all have appropriate names.

In addition, the client and server should not be using the same keystore.